mifosng-android/src/main/java/com/mifos/mifosxdroid/core/MifosBaseActivity.java

```diff
@@ -131,13 +132,21 @@ public void hideKeyboard(View view) {
 
     @Override
     public void logout() {
-        if (PrefManager.getPassCodeStatus()) {
-            startActivity(new Intent(this, PassCodeActivity.class));
-        } else {
-            PrefManager.clearPrefs();
-            startActivity(new Intent(this, SplashScreenActivity.class));
-        }
-        finish();
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setMessage(R.string.logout_message)
+                .setPositiveButton(R.string.action_yes, new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int id) {
+                        PrefManager.clearToken();
```

Comment:
Also, could you double check whether this issue of white text is present on other devices as well? My device is Motorola G5S Plus, Stock Nougat (7.1.1) - API 25.

![pr](https://user-images.githubusercontent.com/12914286/34515538-890d8d64-f098-11e7-9bf9-fad61d57e94b.jpg)

Reply:
@tarun0 I would check this in some other device as well and post the update.

Reply:
@tarun0  I have checked the application on Android 6.0.1,API 23 and on Android 5.0.2 API 21 and I did not find the white text problem.